File: 200_code_python/17.py
# Viết bằng PYTHON
# Tính số Fibonacci thứ n: Nhập số n và in ra số Fibonacci thứ n.

# Nhập số n từ bàn phím
n = int(input("Nhap so n: "))

# Hàm tính Fibonacci bằng cách sử dụng vòng lặp
def fibonacci(n):
    if n <= 0:
        return "So nhap vao phai lon hon 0."
    elif n == 1 or n == 2:
        return 1
    else:
        a, b = 1, 1
        for i in range(3, n + 1):
            a, b = b, a + b
        return b

# In ra số Fibonacci thứ n
print(f"So Fibonacci thu {n} la: {fibonacci(n)}")

File: 200_code_cpp/17.cpp
// Viết bàng C++
// Tính số Fibonacci thứ n: Nhập số n và in ra số Fibonacci thứ n.

#include <iostream>
using namespace std;

// Hàm tính Fibonacci bằng cách sử dụng vòng lập
int fibonacci(int n) {
    if (n <= 0) {
        return -1; // Giá trị không hợp lệ
    } else if (n == 1 || n == 2) 
    {
        return 1;
    } else {
        int a = 1, b = 1, fib = 1;
        for (int i = 3; i <= n; i++) {
            fib = a + b;
            a = b;
            b = fib;
        }
        return fib;
    }   
}

int main() {
    // Nhập số n từ bàn phím
    int n;
    cout << "Nhap so n: ";
    cin >> n;

    // Tính và in ra số Fibonacci thứ n
    int result = fibonacci(n);
    if (result == -1) {
        cout << "So nhap vao phai lon hon 0." << endl;
    } else {
        cout << "So Fibonacci thu " << n << " la: " << result << endl;
    }

    return 0;
}